src/main/java/rx/Single.java

```diff
@@ -2428,6 +2428,32 @@ public void onNext(T t) {
 
         return Observable.create(new OnSubscribeDoOnEach<T>(this.toObservable(), observer)).toSingle();
     }
+
+    @Experimental
+    public final Single<T> doOnEach(final Action1<Notification<? extends T>> onNotification) {
+        if (onNotification == null) {
+            throw new IllegalArgumentException("onNotification is null");
+        }
+
+        Observer<T> observer = new Observer<T>() {
```

Comment:
I just looked at `doOnError` & `doOnSuccess`. If so those should be changed too, right?